### PR TITLE
Put the password generator into a real .sh file

### PR DESCRIPTION
**Pull request by nedbat**
Configuration Pull Request
---

Make sure that the following steps are done before merging

  - [ ] @devops team member has commented with :+1:
  - [ ] are you adding any new default values that need to be overridden when this goes live?  
    - [ ] Open a ticket (DEVOPS) to make sure that they have been added to secure vars.
    - [ ] Add an entry to the CHANGELOG.

### CODE REVIEW

**Comment by e0d** [comment]
@nedbat on master, can update.

**Comment by nedbat** [comment]
@edx/devops @gsong thoughts?

**Comment by nedbat** [comment]
@edx/devops @gsong :ping_pong:

**Review by gsong — APPROVED** [review]

File: util/install/generate-passwords.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+while IFS= read -r line; do
+    # Make a random string. SECRET_KEY's should be longer.
+    length=35
+    if [[ $line == *SECRET_KEY* ]]; then
+        length=100
+    fi
+    REPLACE=$(LC_ALL=C < /dev/urandom tr -dc 'A-Za-z0-9' | head -c$length)
+    # Change "!!null"-to-end-of-line to the password.
+    echo "${line/%!!null*/'$REPLACE'}"
```
Comment (gsong) [inline util/install/generate-passwords.sh]: @nedbat I think the single ticks need to be escaped: `\'$REPLACE\'`

Comment (nedbat) [reply util/install/generate-passwords.sh]: I tested it: with backslashes, the resulting yaml file has backslash-quote.  Without the backslashes, the yaml file correctly has just quotes.

**Comment by nedbat** [comment]
@edx/devops this makes master match the versions of these files in Ficus. Any concerns?

**Comment by gsong** [comment]
Not something we need to address with this specific PR, but will need to revisit at some point is #3716.

**Review by jibsheet — APPROVED** [review]

**Pull request merged**